pkg/controller/inferenceservice/resources/knative/service.go

```diff
@@ -124,6 +124,39 @@ func addLoggerContainerPort(container *v1.Container) {
 	}
 }
 
+func addBatcherAnnotations(batcher *v1alpha2.Batcher, annotations map[string]string) bool {
+	if batcher != nil {
+		annotations[constants.BatcherInternalAnnotationKey] = "true"
+		k_v := map [string] *string {
+			constants.BatcherPortInternalAnnotationKey: batcher.Port,
+			constants.BatcherComponentHostInternalAnnotationKey: batcher.SvcHost,
+			constants.BatcherComponentPortInternalAnnotationKey: batcher.SvcPort,
+			constants.BatcherMaxBatchsizeInternalAnnotationKey: batcher.MaxBatchsize,
+			constants.BatcherMaxLatencyInternalAnnotationKey: batcher.MaxLatency,
+		}
+		for k, v := range k_v {
+			if v != nil {
+				annotations[k] = *v
+			}
+		}
+		return true
+	}
+	return false
+}
+
+func addBatcherContainerPort(container *v1.Container) {
+	if container != nil {
+		if container.Ports == nil {
```

Comment:
Here batcher port is added when container ports is nil right? Does that mean we can not use logger and batcher at the same time?
In fact knative can only expose one user port, before if there's logger then logger port will be exposed. Now from code logger will take priority because it's before batcher?

Reply:
We should be tie it up in pod mutator to ensure logger->batcher->server ?